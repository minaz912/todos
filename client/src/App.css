.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}


/* Style the header */

.header {
  background-color: #f44336;
  padding: 30px 40px;
  /*color: white;*/
  text-align: center;
}


/* Clear floats after the header */

.header:after {
  content: "";
  display: table;
  clear: both;
}

/* Style the "Add" button */

.addBtn {
  padding: 10px;
  font-size: 14px;
  cursor: pointer;
  transition: 0.3s;
}

.todo-container {
  font-size: 16px;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: flex-start;
}

.todo-row {
  display: flex;
  flex-direction: row;
}

.todo-row-first {
  font-weight: 700;
}

.todo-row-second {
  margin-top: 10px;
  text-indent: 10px;
}

.todo-name {
  flex-grow: 1;
}

/* Style the close button */
.icon-right {
  margin: 10px;
  border-radius: 2px;
  padding: 5px;
  color: #fff;
  text-transform: capitalize;
  align-self: flex-end;
}

.icon-close {
  color: #000;
}

.icon-right.icon-close:hover {
  background-color: #f44336;
  color: #fff;
}

.URGENT {
  background-color: #f44336;
}

.IMPORTANT {
  background-color: #ece662;
}

.NORMAL {
  background-color: #5de95d;
}

.add-todo-form input, .add-todo-form textarea, .add-todo-form select {
  display: inline-block;
  width: 75%;
  margin: 10px 0;
  border-radius: 0;
  vertical-align: middle;
}

label {
  display: inline-block;
  width: 25%;
}

select {
  text-align: center;
  background-color: #fff;
  color: #000;
  height: 40px;
}
