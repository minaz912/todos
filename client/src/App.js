import React, { Component } from 'react';
import { connect } from 'react-redux';

import {
  getTodosAction,
  addTodoAction,
  editTodoAction,
  deleteTodoAction
} from './modules/todosActions';

import TodoItem from './modules/todoItem';

import logo from './logo.svg';
import './App.css';

class App extends Component {
  componentDidMount() {
    this.props.getTodosList('ALL');
  }

  getTodos(ev) {
    this.props.getTodosList(ev.target.value);
  }

  addTodo(ev) {
    ev.preventDefault();
    const { title, desc, priority, date } = ev.target;

    const newTodo = {
      name: title.value,
      description: desc.value,
      dueDate: date.value,
      priority: priority.value
    };

    this.props.addTodo(newTodo);
  }

  editTodoCb(
    guid,
    name,
    description,
    priority,
    dueDate,
    completed,
    completionDate
  ) {
    const newCompletionDate = completed
      ? completionDate ? completionDate : new Date()
      : null;
    this.props.editTodo(guid, {
      name,
      description,
      priority,
      dueDate,
      completed,
      completionDate: newCompletionDate
    });
  }

  editTodo(ev) {
    ev.preventDefault();
    const guid = ev.target.todo.value;
    const {title, desc, date, priority} = ev.target;
    const editedTodo = {
      name: title.value,
      description: desc.value,
      dueDate: date.value,
      priority: priority.value
    };
    this.props.editTodo(guid, editedTodo);
  }

  deleteTodo(guid) {
    this.props.deleteTodo(guid);
  }

  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Todos List</h2>
        </div>
        <div className="container">
          <div className="row">
            <div className="view-total col-xs-4 col-xs-offset-6">
              <p>
                <strong>Total:</strong> {this.props.todos.length} Todos
              </p>
            </div>
            <div className="view-mode col-xs-2">
              <select value={this.viewMode} onChange={this.getTodos.bind(this)}>
                <option value="ALL">All</option>
                <option value="TODAY">Today</option>
                <option value="UPCOMING">Upcoming</option>
                <option value="COMPLETED">Completed</option>
              </select>
            </div>
          </div>
          <div className="container">
            <ul id="myUL">
              {this.props.todos.map(todo =>
                <TodoItem
                  key={todo._id}
                  guid={todo._id}
                  title={todo.name}
                  desc={todo.description}
                  dueDate={todo.dueDate}
                  priority={todo.priority}
                  completed={todo.completed}
                  editCb={this.editTodoCb.bind(this)}
                  deleteCb={this.deleteTodo.bind(this)}
                />
              )}
            </ul>
            <div className="header header-red">
              <h2>Add Todo</h2>
              <form
                className="add-todo-form"
                onSubmit={this.addTodo.bind(this)}
              >
                <label htmlFor="title">Title:</label>
                <input
                  className="form-control"
                  type="text"
                  id="title"
                  required
                />

                <br />

                <label htmlFor="desc">Description:</label>
                <textarea
                  className="form-control"
                  rows="6"
                  name="desc"
                  defaultValue="..."
                />

                <br />

                <label htmlFor="date">Due Date:</label>
                <input
                  className="form-control"
                  placeholder="Date"
                  type="date"
                  id="date"
                />

                <br />

                <label htmlFor="priority">Priority:</label>
                <select className="form-control" id="priority">
                  <option value="NORMAL">Normal</option>
                  <option value="IMPORTANT">Important</option>
                  <option value="URGENT">Urgent</option>
                </select>

                <br />

                <label />
                <input
                  type="submit"
                  className="addBtn btn btn-success"
                  value="Add"
                />
              </form>
            </div>
            <div className="header header-yellow">
              <h2>Edit Todo</h2>
              <form
                className="add-todo-form"
                onSubmit={this.editTodo.bind(this)}
              >

                <label htmlFor="todo-select">Todo:</label>
                <select id="todo">
                  {this.props.todos.map((todo) => <option key={todo._id} value={todo._id}>{todo.name || '<Untitled Todo>'}</option>
                  )}
                </select>

                <label htmlFor="title">Title:</label>
                <input
                  className="form-control"
                  type="text"
                  id="title"
                  defaultValue=""
                />

                <br />

                <label htmlFor="desc">Description:</label>
                <textarea
                  className="form-control"
                  rows="3"
                  name="desc"
                  defaultValue="..."
                />

                <br />

                <label htmlFor="date">Due Date:</label>
                <input
                  className="form-control"
                  placeholder="Date"
                  type="date"
                  id="date"
                />

                <br />

                <label htmlFor="priority">Priority:</label>
                <select className="form-control" id="priority">
                  <option value="NORMAL">Normal</option>
                  <option value="IMPORTANT">Important</option>
                  <option value="URGENT">Urgent</option>
                </select>

                <br />

                <label />
                <input
                  type="submit"
                  className="addBtn btn btn-success"
                  value="Add"
                />
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    todos: state.todos
  };
};

const mapDispatchToProps = dispatch => {
  return {
    getTodosList: filter => dispatch(getTodosAction(filter)),
    addTodo: todo => dispatch(addTodoAction(todo)),
    editTodo: (guid, editedTodo) => dispatch(editTodoAction(guid, editedTodo)),
    deleteTodo: guid => dispatch(deleteTodoAction(guid))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
