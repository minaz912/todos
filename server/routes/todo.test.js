import request from 'supertest';

import App from '../index';

test('GET /todos', (done) => {
  request(App)
    .get('/todos')
    .expect('Content-Type', /json/)
    .expect(200)
    .end((err, res) => {
      expect(err).toEqual(null);
      expect(res.data).toMatchSnapshot();
      done();
    })
})

test('POST /todos', (done) => {
  const todo = {
    name: 'Wash the dishes',
    description: 'Use soap to wash the dishes',
    dueDate: new Date(),
    completionDate: new Date(),
    priority: 'URGENT'
  };
  request(App)
    .post('/todos')
    .type('json')
    .set('Accept', 'application/json')
    .expect('Content-Type', /json/)
    .expect(200)
    .send({"todo": todo})
    .end((err, res) => {
      // console.log(res);
      expect(err).toEqual(null);
      expect(res.body.data.name).toEqual(todo.name);
      expect(res.body.data.description).toEqual(todo.description);
      expect(new Date(res.body.data.dueDate)).toEqual(todo.dueDate);
      expect(res.body.data.priority).toEqual(todo.priority);
      done();
    });
})
