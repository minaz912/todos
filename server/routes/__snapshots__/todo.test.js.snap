// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /todos 1`] = `undefined`;
